src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/LobbyManager.java

```diff
@@ -94,8 +97,25 @@ public void clear() {
         lobbyMap.clear();
     }
 
+    @Override
+    public void modifyPlayer(String token, Long lobbyId, String newName, Boolean ready) throws EmptyUsernameException, PlayerNotFoundException, DuplicateUserNameInLobbyException {
+        ILobby lobby = lobbyMap.get(lobbyId);
+        if (newName != null) {
+            if (isEmpty(newName))
+                throw new EmptyUsernameException();
+            lobby.setUserName(token, newName);
+        }
+        if (ready != null) {
+            lobby.setReady(token, ready);
+        }
+    }
+
     @Override
     public Iterator<ILobby> iterator() {
         return lobbyMap.values().iterator();
     }
+
+    private static boolean isEmpty(String s) {
```

Comment:
now we have exactly the same check once in the lobbyManager and once in the lobbyService

maybe we should create some external helper class who does this check @apple00juice

Reply:
Yea we could create a static class, that has all these check methods but more general.